src/path.jl

```diff
@@ -173,11 +173,14 @@ julia> parents(p".")
 """
 function parents(fp::T) where {T <: AbstractPath}
     if hasparent(fp)
-        return [Path(fp, fp.segments[1:i]) for i in 1:length(fp.segments)-1]
-    elseif fp.segments == tuple(".") || isempty(fp.segments)
+        # If we have a root then include that in the returned paths
+        # j = 1 when fp.root is empty and 0 otherwise
+        j = Int(isempty(fp.root))
+        return [Path(fp, fp.segments[1:i]) for i in j:length(fp.segments)-1]
```

Comment:
This code confuses me.
maybe more obvious is is:
```suggestion
        parent_ends = if isempty(fp.root)
            # if no root, then add a empty segment at the start using `fp.segments[1:0]`
            0 : lastindex(fp.segments) - 1
        else
            If there is a root then do not need to add empty segment at start because ...
            1 : lastindex(fp.segments) - 1
        end
        return [Path(fp, fp.segments[1:i]) for i in parent_ends]
```

Reply:
The reason we're doing this is to include the root path (non-empty root with empty segments) when we request all parents. Do you disagree that `/` should be a parent of `/etc`?

Reply:
Oh now i get it,
Can we add a comment being like
```
# absolute paths need empty initial segment at the start, so that one can have just the root
```

Maybe also a clear way to write this:
```suggestion
        parents = [Path(fp, fp.segments[1:i]) for i in 1:length(fp.segments)-1]
        isabs(fp) && pushfirst!(parents, Path(fp, [])  # add root only path 
        return parents
```

Reply:
I've updated the comment to hopefully be a bit more clear. I think it also reads a bit better with the `isrelative` function now.